crates/wysiwyg/src/composer_model/mentions.rs

```diff
@@ -77,7 +77,16 @@ where
         let (start, end) = self.safe_selection();
         let range = self.state.dom.find_range(start, end);
 
-        let new_node = DomNode::new_mention(url, text, attributes);
+        // use the display text or the presence of a `data-mention-type => at-room` attribute to decide the mention type
+        let new_node = if text == "@room".into()
+            || attributes.iter().any(|(k, v)| {
+                k == &S::from("data-mention-type") && v == &S::from("at-room")
+            }) {
+            DomNode::new_at_room_mention(attributes)
+        } else {
+            DomNode::new_mention(url, text, attributes)
```

Comment:
When trying to insert an invalid mention, should the editor still do that? I'm wondering if it could return an error result and/or fall back to `DomNode::new_link()`.

It might not be possible to do without being able to parse mentions but could be worth adding a note

Reply:
That's a good question, and for now I believe we don't have a way of telling if it's valid or not. Will add a note to handle that (after parsing gives us that ability).